client/reader/stream/index.jsx

```diff
@@ -310,6 +312,11 @@ export default connect(
 				recsStreamKey,
 				shouldCombine: shouldCombineCards,
 			} ),
+			postsItems: getTransformedStreamItemsPosts( state, {
+				streamKey,
+				recsStreamKey,
+				shouldCombine: shouldCombineCards,
+			} ),
```

Comment:
Instead of creating a new complex selector, can you reuse the `items` value you already computed? Like:
```js
const items = getTransformedStreamItems( state, ... );
const postsItems = items.map( key => getPostByKey( state, key ) );
return { items, postsItems };
```
Add some memoization and you're done. Another option is a selector that takes `items` as parameter:
```js
const postsItems = getPostsByKey( state, items );
```
This one would be easier to memoize with `createSelector`.

Reply:
In your first example, we'd need to invoke a function in `mapStateToProps`, which isn't desired. Or did you have anything else in mind?

Also, what do you have in mind when you say "some memorization"?

Finally, the last example wouldn't work, because `createSelector` can have only primitive values as arguments AFAIK.

So, could I ask you to elaborate on this one?

Reply:
> In your first example, we'd need to invoke a function in `mapStateToProps`, which isn't desired.

That's a very formal obstacle that can be avoided by extracting the function outside the `connect` call:
```js
const getPostItem = ( state ) => ( key ) => getPostByKey( state, key );
connect( ( state ) => {
  const items = getTransformedStreamItems( state, ... );
  const postsItems = items.map( getPostItem( state ) );
  return { items, postsItems };
} )
```

> Also, what do you have in mind when you say "some memoization"?

To ensure that when `nextItems === prevItems`, then `nextPostsItems === prevPostsItems`. Mere `items.map( ... )` will produce a new array on each state update.

> Finally, the last example wouldn't work, because `createSelector` can have only primitive values as arguments AFAIK.

That can be changed by passing a custom `getCacheKey` function.

But I now found out that there is an existing `getPostsByKeys` selector that already implements it all:
```js
connect( ( state ) => {
  const items = getTransformedStreamItems( state, ... );
  const postsItems = getPostsByKeys( state, items );
  return { items, postsItems };
} )
```